src/OrbitGl/LineGraphTrack.cpp

```diff
@@ -73,40 +96,36 @@ void LineGraphTrack<Dimension>::DrawSeries(PrimitiveAssembler& primitive_assembl
     std::array<float, Dimension> curr_normalized_values =
         GetNormalizedValues(curr_iterator->second, min, inverse_value_range);
 
-    if (is_first) {
-      aggr.StartNewEntry(prev_time, curr_time, curr_normalized_values);
-      is_first = false;
+    if (aggr.GetAccumulatedEntry() == nullptr) {
+      aggr.SetEntry(prev_time, curr_time, curr_normalized_values);
+      continue;
+    }
+    // If the accumulated entry still fits in the pixel
+    if (aggr.GetAccumulatedEntry()->end_tick < next_pixel_start_ns) {
+      // Add the current data to accumulated_entry
+      aggr.MergeDataIntoEntry(prev_time, curr_time, curr_normalized_values);
     } else {
-      // If the size of the current entry is less than a pixel
-      if (aggr.GetEntry().end_tick < next_pixel_start_ns) {
-        // Add the current data to accumulated_entry
-        aggr.AppendData(prev_time, curr_time, curr_normalized_values);
-      } else {
-        // Otherwise, draw the accumulated_entry and start accumulating a new one
-        DrawSingleSeriesEntry(primitive_assembler, aggr.GetEntry().start_tick,
-                              aggr.GetEntry().end_tick, prev_normalized_values,
-                              aggr.GetEntry().values, z, false);
-
-        next_pixel_start_ns = orbit_client_data::GetNextPixelBoundaryTimeNs(
-            aggr.GetEntry().end_tick, resolution_in_pixels, min_tick, max_tick);
-
-        prev_normalized_values = aggr.GetEntry().values;
-        aggr.StartNewEntry(prev_time, curr_time, curr_normalized_values);
-      }
+      auto accumulated_entry = *aggr.GetAccumulatedEntry();
+      // Otherwise, draw the accumulated_entry and start accumulating a new one
+      draw_aggregated(accumulated_entry, false);
+
+      next_pixel_start_ns = orbit_client_data::GetNextPixelBoundaryTimeNs(
+          aggr.GetAccumulatedEntry()->end_tick, resolution_in_pixels, min_tick, max_tick);
+
+      aggr.SetEntry(prev_time, curr_time, curr_normalized_values);
```

Comment:
Nit: Just an OTC. I prefer to update `next_pixel_start_ns` at the end.

```
aggr.SetEntry(prev_time, curr_time, curr_normalized_values);
next_pixel_start_ns = orbit_client_data::GetNextPixelBoundaryTimeNs(
          aggr.GetAccumulatedEntry()->end_tick, resolution_in_pixels, min_tick, max_tick);
```

Reply:
It must be done before - clarified with a comment